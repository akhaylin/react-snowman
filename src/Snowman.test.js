import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Snowman from './Snowman';
import * as random from './words.js'
random.randomWord = jest.fn()

test('game ends after MaxWrong guesses', () => {
  random.randomWord.mockReturnValue("apple")
  const { container } = render(<Snowman maxWrong={6} />);

  fireEvent.click(container.querySelector('button[value="z"]'));
  fireEvent.click(container.querySelector('button[value="x"]'));
  fireEvent.click(container.querySelector('button[value="v"]'));
  fireEvent.click(container.querySelector('button[value="k"]'));
  fireEvent.click(container.querySelector('button[value="j"]'));
  fireEvent.click(container.querySelector('button[value="q"]'));

  // After 6 wrong guesses, "You lose" message should appear
  expect(container.querySelector('.Snowman')).toHaveTextContent('You lose!');

  // Check that buttons aren't visible anymore by trying to access one and expecting it to be null
  expect(container.querySelector('button[value="a"]')).not.toBeInTheDocument();
  expect(container.querySelector('button[value="b"]')).not.toBeInTheDocument();


  // Check if the correct word is displayed
  expect(container.querySelector('.Snowman-correctWord')).toBeInTheDocument('apple');
})

it("matches snapshot", function() {
  random.randomWord.mockReturnValue("apple")
  const { container } = render(<Snowman maxWrong={6} />);

  fireEvent.click(container.querySelector('button[value="z"]'));
  fireEvent.click(container.querySelector('button[value="x"]'));
  fireEvent.click(container.querySelector('button[value="v"]'));
  fireEvent.click(container.querySelector('button[value="k"]'));
  fireEvent.click(container.querySelector('button[value="j"]'));
  fireEvent.click(container.querySelector('button[value="q"]'));

  expect(container).toMatchSnapshot();
});