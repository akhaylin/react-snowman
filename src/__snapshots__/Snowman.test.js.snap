// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p
      class="Snowman-wrongGuesses"
    >
      Number wrong: 
      6
    </p>
    <p>
      You lose!
    </p>
    <p
      class="Snowman-correctWord"
    >
      The correct word was: 
      apple
    </p>
    <div>
      <button>
        Restart
      </button>
    </div>
  </div>
</div>
`;
